home/private_dot_local/lib/agent-equipment/agent_equipment/authoring.py

```
@@ -0,0 +1,1165 @@
+
+
+@dataclass(frozen=True, slots=True)
+class AuthoringError:
+    """One stable, redacted failure from an authored proposal command."""
+
+    code: str
+    message: str
+
+    def __post_init__(self) -> None:
+        if re.fullmatch(r"[A-Z][A-Z0-9_]*", self.code) is None:
+            raise ValueError("authoring error codes must be stable identifiers")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)